### PR TITLE
Add icon badge in Chromium

### PR DESCRIPTION
**Pull request by koops76**
Example of icon badge

### CODE REVIEW

**Comment by J0WI** [comment]
ping @Hainish

File: chromium/background.js [hunk 1]
```diff
@@ -126,6 +147,12 @@ function updateState() {
     chrome.browserAction.setTitle({
       title: "HTTPS Everywhere (" + iconState + ")"
     });
+
+    chrome.browserAction.setBadgeBackgroundColor({color: "#00cc00"});
+
+    const showBadge = activeCount > 0 && isExtensionEnabled;
+
```
Comment (Hainish) [inline chromium/background.js]: Remove whitespace from this line.

File: chromium/background.js [hunk 2]
```diff
@@ -191,6 +218,8 @@ AppliedRulesets.prototype = {
       this.active_tab_rules[tabId] = {};
       this.active_tab_rules[tabId][ruleset.name] = ruleset;
     }
+
+    updateState();
```
Comment (Hainish) [inline chromium/background.js]: I think it makes more sense to place this on what is currently line 325 of this file, so as to eliminate unnecessary calls.

**Comment by koops76** [comment]
@Hainish Done.

**Comment by koops76** [comment]
@Hainish Will do so in a separate PR.

**Comment by Hainish** [comment]
There appears to be a change in the `translations` submodule?

**Comment by koops76** [comment]
@Hainish Maybe I accidentally updated it.

**Comment by Hainish** [comment]
@koops76 do you plan on updating this PR or should I close it?

**Comment by koops76** [comment]
@Hainish I already fixed all issues with my PR. Do you want me to add the menu option too?

**Comment by Hainish** [comment]
@koops76 this needs to be fixed:

> Submodule translations updated from 020039 to 303723

And yes, if you could add the menu option.

**Comment by koops76** [comment]
@Hainish Done.

**Comment by Hainish** [comment]
Great, last thing is the menu option.

**Comment by koops76** [comment]
@Hainish Please merge this unless there is a serious issue preventing this from being merged. Especially since soon Firefox will use the WebExtensions version and the legacy version used to have the counter.

**Comment by Hainish** [comment]
@koops76 https://github.com/EFForg/https-everywhere/pull/11800 migrates the XPCOM option to show a counter into WebExtensions' storage as `showCounter`.  We've also added an option UI.  Having a toggle for the counter is a requirement for deployment, as I can see users having a strong preference either way.  The best place for this is in the options dialog.

**Comment by koops76** [comment]
@Hainish Should I add it into `popup.html`?

**Comment by Hainish** [comment]
@koops76 either the popup or the options dialog is fine.  I might move it around later if I think it looks more appropriate in a different location.

**Comment by koops76** [comment]
@Hainish Take a look.

**Comment by koops76** [comment]
I will continue my work on exporting/importing settings in this PR since it depends on the code I have put into it.

**Comment by Hainish** [comment]
@koops76 thanks, let me know when this is in a good place to review.  When you think it is, it's helpful for me if you stop making any more commits in this branch so I have a solid, unchanging diff to do code review from.

**Comment by Hainish** [comment]
@koops76 is this ready for review?  See above comment

**Comment by Hainish** [comment]
As it stands this PR introduces a lot of non-orthogonal code, completely unrelated to the stated objective.  This belongs in another PR, which can be reviewed and merged separately without a pressing deadline.

This release is going out today or tomorrow, and it seems this PR is not ready and I will not have the opportunity to give it a proper review before tomorrow's launch.  As it stands this functionality will have to wait until next release.

**Comment by koops76** [comment]
@Hainish OK, I am going to close this and continue working on this branch. Then I will separate changes into their own PRs.

**Comment by Hainish** [comment]
@koops76 any progress on this?  It would be great to get into the next version of HTTPS Everywhere.  If you submit a PR which adds the badge and reads the storage var for whether it's disabled or not, I can take care of the options page.

**Pull request closed without merging**